src/BlazorSignalR/Internal/BlazorHttpConnectionFactory.cs

```diff
@@ -13,22 +15,35 @@ internal class BlazorHttpConnectionFactory : IConnectionFactory
     {
         private readonly BlazorHttpConnectionOptions _options;
         private readonly IJSRuntime _jsRuntime;
+        private readonly IUriHelper _uriHelper;
+        private readonly bool _isServerSide;
         private readonly ILoggerFactory _loggerFactory;
 
-        public BlazorHttpConnectionFactory(IOptions<BlazorHttpConnectionOptions> options, IJSRuntime jsRuntime, ILoggerFactory loggerFactory)
+        public BlazorHttpConnectionFactory(
+            IOptions<BlazorHttpConnectionOptions> options,
+            IJSRuntime jsRuntime,
+            IUriHelper uriHelper,
+            ILoggerFactory loggerFactory)
         {
             if (jsRuntime == null)
                 throw new ArgumentNullException(nameof(jsRuntime));
 
+            if (uriHelper == null)
+                throw new ArgumentNullException(nameof(uriHelper));
+
             _options = options.Value;
             _jsRuntime = jsRuntime;
+            _uriHelper = uriHelper;
             _loggerFactory = loggerFactory;
+
+            // TODO: Is there a better/cleaner way?
+            _isServerSide = AppDomain.CurrentDomain.GetAssemblies().Any(p => p.GetName().Name == "Microsoft.AspNetCore.Mvc");
```

Comment:
As we already have an instance of the js runtime, with something like
```suggestion
            _isServerSide = !(_jsRuntime is JSInProcessRuntime);
```

Reply:
I do not think using reflection just incase webworker etc gets added in the near future in worthwhile as the Reflection code is more likely to break without anyone noticing. Plus such features are a while away.

Reply:
Ok, then I will use this for now with a comment that this may break things if WebWorkers and the like is ever supported.